plugin/role_set_utils.go

```diff
@@ -0,0 +1,165 @@
+package gcpsecrets
```

Comment:
as a note, ideally i'll move the rest of the shared utils from role_set into role_set utils so roleset only includes the Roleset related types and methods on those types